drivers/hwinfo/Kconfig

```diff
@@ -0,0 +1,35 @@
+# Kconfig -HW Info driver configuration options
+
+#
+# Copyright (c) 2019 Alexander Wachter
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig DEVICE_ID
+	bool "Device id driver"
+	help
+	  Enable device id driver.
+
+if DEVICE_ID
+
+config DEVICE_ID_NAME
+	string "Device ID Name"
+	default "DEV_ID"
+	help
+	  Specify the device name to be used for the ID driver.
+
+config DEVICE_ID_STM32
+	bool "STM32 device ID"
+	depends on (SOC_SERIES_STM32F0X || SOC_SERIES_STM32F3X || \
+		    SOC_SERIES_STM32L4X || SOC_SERIES_STM32F7X)
+	help
+	  Enable STM32 Device ID driver.
+
+config DEVICE_ID_NRF
+	bool "NRF device ID"
+	depends on (SOC_SERIES_NRF52X || SOC_SERIES_NRF51X)
```

Comment:
I don't know how you would do it (maybe just `default y`), but the appropriate `DEVICE_ID_foo` should be default-selected.  It's annoying to set `CONFIG_DEVICE_ID=y` then have the runtime lookup fail because you didn't select an implementation.

Reply:
Done